wp/plugin/lib/equality.ml

```diff
@@ -0,0 +1,387 @@
+open !Core_kernel
+open Bap.Std
+open Graphlib.Std
+open Base__Option.Monad_infix
+
+(* some of these pattern matchings are massive; we don't want fragile warnings *)
+[@@@warning "-4"]
+
+module H = Bap.Std.Graphs.Ir
+module J = Graphlib.To_ocamlgraph (H)
+module BFS = Graph.Traverse.Bfs (J)
+module TidSet = Tid.Set
+
+module TidMap = Tid.Map
+module VarMap = Var.Map
+module VarSet = Var.Set
+module TidTupleMap = Map.Make(struct type t = Tid.t * Tid.t [@@deriving sexp, compare] end)
+
+type varToVarMap = Var.t VarMap.t
+
+(* Replaces the real variables with indices with only their base *)
+let rec strip_indices (e : Exp.t) : Exp.t =
```

Comment:
I think this could be expressed more succinctly using the `mapper` interface. I think this is mapping Var.base over Var constructors and that's it?